src/test_reference.jl

```diff
@@ -45,6 +93,12 @@ cases.
 # Images can also be stored as hash. Note however that this
 # can only check for equality (no tolerance possible)
 @test_reference "references/camera.sha256" testimage("cameraman")
+
+# test images using ImageQualityIndexes.PSNR
+@test_reference "references/camera.png" testimage("cameraman") by=(x,y)->:psnr(x,y)>25
```

Comment:
Is it possible to skip `:` here in the macro? For example,
```julia
@test_reference "references/camera.png" testimage("cameraman") by=(x,y)->psnr(x,y)>25
```

Reply:
hmm should be. Lets fix that.
Seems like maybe you are missing an `esc`?

Reply:
:( Unfortunately,  I'm not familiar with macros, can you make a patch to it?

Looks like we need to rewrite the `@test_reference` so that `psnr` is evaluated before passing to `test_reference`.

https://github.com/Evizero/ReferenceTests.jl/blob/41a61f2c05d96ea5a97443eae87a057146957377/src/test_reference.jl#L50-L58

Reply:
I think it just needs to be
```
 macro test_reference(reference, actual, kws...) 
     dir = Base.source_dir() 
     expr = :(test_reference(abspath(joinpath($dir, $(esc(reference)))), $(esc(actual)))) 
     for kw in kws 
         (kw isa Expr && kw.head == :(=)) || error("invalid signature for @test_reference") 
         k, v = kw.args
         push!(expr.args, Expr(:kw, k, esc(v))) 
     end 
     expr 
 end 
```